File: BubbleSort/BubbleSort.rb
n = gets.strip.to_i
a = gets.strip
a = a.split(' ').map(&:to_i)

total_number_of_swaps = 0
a.each do |el|
    number_of_swaps = 0
    idx = 0
    while(idx < a.length - 1)
        if(a[idx] > a[idx + 1])
            a[idx], a[idx + 1] = a[idx + 1], a[idx]
            number_of_swaps += 1
            total_number_of_swaps += 1
        end
        idx += 1
    end
    if(number_of_swaps == 0)
        break
    end
end

puts "Array is sorted in #{total_number_of_swaps} swaps."
puts "First Element: #{a[0]}"
puts "Last Element: #{a[-1]}"

File: BubbleSort/BubbleSort.py
n = int(raw_input().strip())
a = map(int, raw_input().strip().split(' '))
tot = 0
for i in range(n):
    cnt = 0
    for j in range(1,n):
        if a[j-1] > a[j]:
            cnt += 1
            a[j-1],a[j]=a[j],a[j-1]
    if cnt==0:
        break
    tot += cnt
print 'Array is sorted in %d swaps.' % tot
print 'First Element: %d' % a[0]
print 'Last Element: %d' % a[-1]